test/models/location_test.rb

```diff
@@ -3,4 +3,64 @@
 
 class LocationTest < ActiveSupport::TestCase
   include TaxonomiesBaseTest
+  # List of valid location names.
+  def valid_loc_name_list
+    [
+      RFauxFactory.gen_alpha(1),
+      RFauxFactory.gen_alpha(246),
+      *RFauxFactory.gen_strings(1..246, exclude: [:html]).values,
+      RFauxFactory.gen_html(rand((1..221)))
+    ]
+  end
+
+  # List of invalid location names.
+  def invalid_loc_name_list
+    [
+      '',
+      ' ',
+      "\t",
+      *RFauxFactory.gen_strings(247).values
+    ]
+  end
+
+  test 'should create with multiple valid names' do
+    valid_loc_name_list.each do |name|
+      location = FactoryBot.build(:location, :name => name)
+      assert location.valid?, "Can't create location with valid name #{name}"
+    end
+  end
+
+  test 'should not create with multiple invalid names' do
+    invalid_loc_name_list.each do |name|
+      location = FactoryBot.build(:location, :name => name)
+      refute location.valid?, "Can create location with invalid name #{name}"
+    end
+  end
+
+  test 'should update with multiple valid names' do
+    location = FactoryBot.create(:location)
+    valid_loc_name_list.each do |name|
+      location.name = name
+      assert location.valid?, "Can't update location with valid name #{name}"
+      assert_equal location.name, name
```

Comment:
this is redundant check, the attribute does not change, if you'd want to check it's really persisted in DB, you'd need to call save instead of valid? and then reload the location object, in this case I think checking that `valid?` is true on line above is enough and I'd drop this line